### PR TITLE
Added support to install rhods on PSI with git repo

### PR DESCRIPTION
**Pull request by tarukumar**
./run_robot_test.sh --include install --test-variable operator_version:v1140-2 --test-variable cluster_type:PSI --test-case tasks/Tasks/rhods_olm.robot
Signed-off-by: Tarun Kumar <takumar@redhat.com>

### CODE REVIEW

**Comment by tarukumar** [comment]
`Rhods Olm :: Perform and verify RHODS OLM tasks                               
==============================================================================
Can Install RHODS Operator                                            Cluster type: PSI
.Installing RHODS operator in PSI
.Verifying RHODS installation

[ WARN ] Timeout - 4 pods found:                                              

jupyterhub-1-59rng
jupyterhub-1-n5wtf
jupyterhub-1-vsnc4
jupyterhub-db-1-55vzw
Can Install RHODS Operator                                            Verifying Builds
Builds Verified
[ WARN ] Timeout - Pods with containers not ready:                            

Name: jupyterhub-1-n5wtf
Status: Running
Reason: 
Message: 
Conditions: [{'type': 'Initialized', 'status': 'True', 'lastProbeTime': None, 'lastTransitionTime': '2022-07-26T07:57:25Z'}, {'type': 'Ready', 'status': 'False', 'lastProbeTime': None, 'lastTransitionTime': '2022-07-26T07:55:25Z', 'reason': 'ContainersNotReady', 'message': 'containers with unready status: [jupyterhub]'}, {'type': 'ContainersReady', 'status': 'False', 'lastProbeTime': None, 'lastTransitionTime': '2022-07-26T07:55:25Z', 'reason': 'ContainersNotReady', 'message': 'containers with unready status: [jupyterhub]'}, {'type': 'PodScheduled', 'status': 'True', 'lastProbeTime': None, 'lastTransitionTime': '2022-07-26T07:55:24Z'}]

Name: jupyterhub-1-vsnc4
Status: Running
Reason: 
Message: 
Conditions: [{'type': 'Initialized', 'status': 'True', 'lastProbeTime': None, 'lastTransitionTime': '2022-07-26T07:57:25Z'}, {'type': 'Ready', 'status': 'False', 'lastProbeTime': None, 'lastTransitionTime': '2022-07-26T07:55:25Z', 'reason': 'ContainersNotReady', 'message': 'containers with unready status: [jupyterhub]'}, {'type': 'ContainersReady', 'status': 'False', 'lastProbeTime': None, 'lastTransitionTime': '2022-07-26T07:55:25Z', 'reason': 'ContainersNotReady', 'message': 'containers with unready status: [jupyterhub]'}, {'type': 'PodScheduled', 'status': 'True', 'lastProbeTime': None, 'lastTransitionTime': '2022-07-26T07:55:24Z'}]

Can Install RHODS Operator                                            Verified redhat-ods-applications
Verified redhat-ods-operator
Verified redhat-ods-monitoring
RHODS has been installed
Can Install RHODS Operator                                            | PASS |
------------------------------------------------------------------------------
Rhods Olm :: Perform and verify RHODS OLM tasks                       | PASS |
1 task, 1 passed, 0 failed
==============================================================================
Output:  /home/takumar/Development/ods-ci/test-output/ods-ci-2022-07-26-13-15-wWJ0SwRBAv/output.xml
XUnit:   /home/takumar/Development/ods-ci/test-output/ods-ci-2022-07-26-13-15-wWJ0SwRBAv/xunit_test_result.xml
Log:     /home/takumar/Development/ods-ci/test-output/ods-ci-2022-07-26-13-15-wWJ0SwRBAv/log.html
Report:  /home/takumar/Development/ods-ci/test-output/ods-ci-2022-07-26-13-15-wWJ0SwRBAv/test_report.html`

File: tasks/Resources/RHODS_OLM/install/install.robot [hunk 1]
```diff
@@ -4,10 +4,16 @@ Installing RHODS Operator ${operator_version}
...
+
 RHODS Operator Should Be installed
   Verify RHODS Installation
   ${version} =  Get RHODS Version
   Set Global Variable  ${RHODS_VERSION}  ${version}
-  Log  RHODS has been installed  console=yes
\ No newline at end of file
+  Log  RHODS has been installed  console=yes
+
+Install Teardonw
```
Comment (mwaykole) [inline tasks/Resources/RHODS_OLM/install/install.robot:16]: typeo

Comment (tarukumar) [reply tasks/Resources/RHODS_OLM/install/install.robot]: ack

**Review by lugi0 — APPROVED** [review]
lgtm

File: tasks/Resources/RHODS_OLM/install/oc_install.robot [hunk 2]
```diff
@@ -52,12 +64,23 @@ Verify Builds Number
...
-    Should Be Equal As Strings  ${build}[status][phase]  ${build_status}  
+    Should Be Equal As Strings  ${build}[status][phase]  ${build_status}
     Should Not Be Equal As Strings  ${build}[status][phase]  Cancelled
     Should Not Be Equal As Strings  ${build}[status][phase]  Failed
     Should Not Be Equal As Strings  ${build}[status][phase]  Error
-  END
\ No newline at end of file
+  END
+
+Install RHODS on OSD Cluster
```
Comment (pablofelix) [inline tasks/Resources/RHODS_OLM/install/oc_install.robot:77]: The catalogsource installation is the same in PSI/OSD

**Review by pablofelix — DISMISSED** [review]

File: tasks/Resources/RHODS_OLM/install/oc_install.robot (hunk 2 above)
Comment (tarukumar) [reply tasks/Resources/RHODS_OLM/install/oc_install.robot]: I am aware of that just that I didn't want to disturb what was already present so i added the condition. However i removed the OSD part

File: tasks/Resources/RHODS_OLM/install/install.robot [hunk 3]
```diff
@@ -4,10 +4,16 @@ Installing RHODS Operator ${operator_version}
   Run Keyword Unless  ${is_operator_installed}  Run Keywords
   ...  Log  Installing RHODS operator in ${cluster_type}  console=yes  AND
   ...  Set Suite Variable  ${operator_version}  AND
-  ...  Install RHODS  ${operator_version}
-    
+  ...  Set Global Variable  ${RHODS_INSTALL_REPO}  AND
```
[the remark this answers is not in the document]
Comment (tarukumar) [reply tasks/Resources/RHODS_OLM/install/install.robot]: The idea was if anyone want to use the variable from other suite they could have used it. Thinking back i think it is good idea to reduce the scope. I have reduce the scope to only test case.

File: tasks/Resources/RHODS_OLM/install/oc_install.robot [hunk 4]
```diff
@@ -1,33 +1,45 @@
...
+           Set Local Variable    ${operator_url}        quay.io/modh/qe-catalog-source:${operator_version}
+      ELSE
+           Should Start With      ${operator_version}     quay.io     msg=you should provide the full build link
+           Set Local Variable    ${operator_url}        ${operator_version}
+      END
+      ${data}     Split String    ${RHODS_INSTALL_REPO}     /
+      ${filename}  Split String     ${data}[-1]            .
+      Set Test Variable     ${filename}       ${filename}[0]
+      Run    git clone ${RHODS_INSTALL_REPO}
+      Run    cd ${EXECDIR}/${filename} && ./setup.sh ${operator_url} > ${EXECDIR}/olm.txt
```
[the remark this answers is not in the document]
Comment (tarukumar) [reply tasks/Resources/RHODS_OLM/install/oc_install.robot]: Ack, i have added the code to check rc status

**Review by pablofelix — APPROVED** [review]

**Review by mwaykole — APPROVED** [review]

**Pull request merged**